### ISSUE TITLE
Searching for a string returns an empty object

### ISSUE THREAD

**Issue by jhirbour** [ContinuityControl/ncua#5]
Searching for a string returns an empty object. I'd expect it to throw an Institution not found error or something similar.

```
2.2.3 :079 > pp NCUA.find_credit_union("asdf")
gems/ruby-2.2.3@control/gems/ncua-0.8.3/lib/ncua/credit_union/scraper.rb:10: warning: wrong element type nil at 16 (expected array)
gems/ruby-2.2.3@control/gems/ncua-0.8.3/lib/ncua/credit_union/scraper.rb:10: warning: ignoring wrong elements is deprecated, remove them explicitly
gems/ncua-0.8.3/lib/ncua/credit_union/scraper.rb:10: warning: this causes ArgumentError in the next release
#<NCUA::CreditUnion::Details:0x00000010dbbdb0
 @attributes=
  {:credit_union_name=>"",
   :charter_number=>"",
   :credit_union_type=>"",
   :credit_union_status=>"",
   :corporate_credit_union=>"",
   :credit_union_charter_year=>"",
   :current_charter_issue_date=>"",
   :date_insured=>"",
   :charter_state=>"",
   :region=>"",
   :field_of_membership_type=>"",
   :low_income_designation=>"",
   :member_of_fhlb=>"",
   :assets=>"-$1",
   :peer_group=>"",
   :number_of_members=>"-1",
   :address=>"",
   :city_state_zip_code=>",",
   :country=>"",
   :county=>"",
   :phone=>"",
   :website=>"",
```

I've coded around this currently by doing this in my local app.

```
institution = ::NCUA.find_credit_union(charter_number)
raise InstitutionNotFound if institution.charter_number == ""
```

**Comment by treznick** [comment]
I'll PR a change to deal with this

**Comment by danbernier** [comment]
Could be as easy as raising n argument exception...
On Nov 14, 2015 1:12 AM, "Tom Reznick" notifications@github.com wrote:

> I'll PR a change to deal with this
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ContinuityControl/ncua/issues/5#issuecomment-156644307
> .

## 

## 

---

This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.

**Comment by treznick** [comment]
I'm not sure it's that easy though. Here's the relevant bit of code from the `DetailsClient`

``` ruby
def get_details(charter_number)
  charter_number = charter_number.to_s.strip
  if charter_number.empty?
    raise ArgumentError, "charter number cannot be nil or empty string"
  end
  #...
end
```

Do we check that a charter_number isn't a string? The string `"123456"` is technically valid insofar as the HTTP request is concerned. Do we try to cast to an int to check it?

Moreover do we know all of the ways that this will fail? We do know what the failure looks like - a record without a charter number. My hunch (but it's only a hunch, and I'm PR commenting before coffee now, so there's that) is that we might be safer if we raise if we get a record without a charter number. Which kind of makes sense given that this whole bit of code is trying to get directly to a credit union. 

Eh, I could go either way here. What think you @danbernier ?

**Comment by danbernier** [comment]
True, we can raise if the response looks hinkey. But we can raise before we
call, if the value looks wrong: if it's a string, it should be non-empty
and numeric.
On Nov 14, 2015 10:29 AM, "Tom Reznick" notifications@github.com wrote:

> I'm not sure it's that easy though. Here's the relevant bit of code from
> the DetailsClient
> 
> def get_details(charter_number)
>   charter_number = charter_number.to_s.strip
>   if charter_number.empty?
>     raise ArgumentError, "charter number cannot be nil or empty string"
>   end
>   #...end
> 
> Do we check that a charter_number isn't a string? The string "123456" is
> technically valid insofar as the HTTP request is concerned. Do we try to
> cast to an int to check it?
> 
> Moreover do we know all of the ways that this will fail? We do know what
> the failure looks like - a record without a charter number. My hunch (but
> it's only a hunch, and I'm PR commenting before coffee now, so there's
> that) is that we might be safer if we raise if we get a record without a
> charter number. Which kind of makes sense given that this whole bit of code
> is trying to get directly to a credit union.
> 
> Eh, I could go either way here. What think you @danbernier
> https://github.com/danbernier ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ContinuityControl/ncua/issues/5#issuecomment-156712295
> .

## 

## 

---

This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.

**Comment by treznick** [comment]
That makes sense

On Saturday, November 14, 2015, Dan Bernier notifications@github.com
wrote:

> True, we can raise if the response looks hinkey. But we can raise before we
> call, if the value looks wrong: if it's a string, it should be non-empty
> and numeric.
> On Nov 14, 2015 10:29 AM, "Tom Reznick" <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> 
> > I'm not sure it's that easy though. Here's the relevant bit of code from
> > the DetailsClient
> > 
> > def get_details(charter_number)
> > charter_number = charter_number.to_s.strip
> > if charter_number.empty?
> > raise ArgumentError, "charter number cannot be nil or empty string"
> > end
> > #...end
> > 
> > Do we check that a charter_number isn't a string? The string "123456" is
> > technically valid insofar as the HTTP request is concerned. Do we try to
> > cast to an int to check it?
> > 
> > Moreover do we know all of the ways that this will fail? We do know what
> > the failure looks like - a record without a charter number. My hunch (but
> > it's only a hunch, and I'm PR commenting before coffee now, so there's
> > that) is that we might be safer if we raise if we get a record without a
> > charter number. Which kind of makes sense given that this whole bit of
> > code
> > is trying to get directly to a credit union.
> > 
> > Eh, I could go either way here. What think you @danbernier
> > https://github.com/danbernier ?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/ContinuityControl/ncua/issues/5#issuecomment-156712295>
> > .
> 
> ## 
> 
> ## 
> 
> ---
> 
> This email and all files transmitted with it are intended solely for use
> by
> the intended recipient. Any other use is prohibited. If you received this
> email in error, please notify the sender, delete the email and destroy all
> attachments. Opinions and ideas expressed herein are solely those of the
> sender, and may not reflect the official position of Continuity.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ContinuityControl/ncua/issues/5#issuecomment-156719830
> .

## 

Sent from my phone. Please excuse brevity and errors.

## 

## 

---

This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.

**Comment by treznick** [comment]
Fixed in `0.9.0`
